monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/commons/DelegatesUtil.java

```diff
@@ -76,8 +98,80 @@ public static Artifacts getArtifactsFromMI(String groupId, List<String> nodeList
                 }
             }
         }
+        
+        //ordering   
+        Comparator<ArtifactsInner> comparatorObject;
+        switch (orderBy.toLowerCase()) {
+            //for any other ordering options
+            default: comparatorObject = Comparator.comparing(ArtifactsInner::getNameIgnoreCase); break;
+        }
+        if ("desc".equalsIgnoreCase(order)) {
+            Collections.sort(artifacts, comparatorObject.reversed());
+        } else {
+            Collections.sort(artifacts, comparatorObject);
+        }  
+
         return artifacts;
     }
+    
+    public static ArtifactsResourceResponse getPaginatedArtifactResponse(String groupId, List<String> nodeList, 
+        String artifactType, String searchKey, String lowerLimit, String upperLimit, String order, String orderBy,
```

Comment:
why specify variable type each time?
String artifactType, searchKey, lowerLimit ...

Reply:
in the parameter list it's not allowed right?